File: Extra/leetcode/3.LongestSubstringWithoutRepeatingCharacters/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
  public int lengthOfLongestSubstring(String s) {
    int best = 0;
    int curr = 0;
    
    for (int i = 0; i < s.length(); i++) {
      List<Character> lst = new ArrayList<>();
      String newstring = s.substring(i);
      for (int j = 0; j < newstring.length(); j++) {
        char c = newstring.charAt(j);
        if (lst.contains(c)) {
          if (best < curr) best = curr;
          curr = 0;
          break;
        } else {
          lst.add(c);
          curr++;
        }
      }
      if (curr > best) best = curr;
      curr = 0;
    } 
    return best;
  }
}

File: Extra/leetcode/3.LongestSubstringWithoutRepeatingCharacters/solution.py
class Solution:
    
  
  
  def lengthOfLongestSubstring(self, s: str) -> int:
    def aux(s: str)  -> int:
      best = 0
      cont = 0
      curr = []
      for c in s:
        #print(c + " ")
        #print(curr)
        if c in curr:
          if cont > best: best = cont 
          cont = 1
          if curr[len(curr)-2] == c and curr[len(curr)-1] != c : 
            temp = curr[len(curr)-1]
            curr.clear()
            curr.append(temp)
            cont += 1
          else :
            curr.clear()
          curr.append(c)
        else:
          cont += 1
          curr.append(c)
      if cont > best: return cont
      return best

    order = aux(s)
    rev = aux(s[::-1])
    if order > rev: return order
    return rev
    